File: Python/Base 7.py
class Solution(object):
    def convertToBase7(self, num):
        """
        :type num: int
        :rtype: str
        """
        s = ''
        flag = '' if num >= 0 else '-'
        num = abs(num)
        if num == 0:
            return '0'
        while num > 0:
            s += str(num % 7)
            num /= 7
        return flag + s[::-1]

File: cpp/Base 7.cpp
class Solution {
public:
    string convertToBase7(int num) {

        if (num == 0) {
            return string(1, '0');
        }

        bool flag;
        flag = num < 0;
        num = abs(num);
        vector<int> result;
        while (num > 0) {
            result.push_back(num % 7);
            num /= 7;
        }
        string string1;
        if (flag) {
            string1.push_back('-');
        }
        for (auto i = result.rbegin(); i != result.rend(); i++) {
            string1 += to_string(*i);
        }
        return string1;
    }
};
